solutions/standard/main.tf

```diff
@@ -49,13 +51,68 @@ module "kms" {
   ]
 }
 
+#######################################################################################################################
+# COS
+#######################################################################################################################
+
+locals {
+  cos_kms_key_crn   = var.existing_cos_bucket_name != null ? null : var.existing_kms_root_key_id != null ? var.existing_kms_root_key_id : module.kms[0].keys[format("%s.%s", var.en_key_ring_name, var.en_key_name)].crn
```

Comment:
ok so this is where things get a little messy. You are using the same KMS key that is used for Event Notification encryption. We actually will want new KMS variables for the COS bucket here too. So it should create its own key ring and key for COS, or optionally allow user to supply an existing KMS key to encrypt the bucket.